R/as-data-frame.R

```diff
@@ -89,17 +98,20 @@ as.data.frame.CrunchDataset <- function (x, row.names = NULL, optional = FALSE,
 #' @rdname dataset-to-R
 #' @export
 as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE, ...) {
-    x <- x$.crunchDataset
+    ds <- attr(x, "crunchDataset")
     default.stringsAsFactors <- function () FALSE
     limit <- min(c(10000, getOption("crunch.data.frame.limit")))
-    if (nrow(x) * ncol(x) > limit) {
+    if (nrow(ds) * ncol(ds) > limit) {
         ## TODO: switch to downloading CSV and reading that?
```

Comment:
I agree, though in the interest of moving forward on other projects, and since this is a pretty large PR already. I'm going to vote on putting this as a new issue/improvement to be implemented in a different PR. It would be good to have a clean DL csv/read into dataframe within `as.data.frame` as well as as a separate function.